### ISSUE TITLE
[Bug]: Keypad Recent Transactions Not Loading If Logged Out

### ISSUE THREAD

**Issue by leesalminen** [btcpayserver/btcpayserver#5530]
### What is your BTCPay version?

main as of 2023-12-03 (commit hash 60ff7e86b8795601b72e5454e2adcbbd110184aa)

### How did you deploy BTCPay Server?

manual

### What happened?

While testing out the new recent transactions log on the Keypad screen (#5478), I noticed that if you aren't currently logged in, the recent transactions never load. Also, the loading spinner never goes away and there is never any indication to the user that they must be logged in to view.

### How did you encounter this bug?

1.) Log out of BTCPayServer
2.) Navigate to a Point of Sale Keypad
3.) Click the recent transactions button
4.) Loading spinner never stops

### What browser do you use?

Safari 17.1.2

### Additional information

<img width="608" alt="Screenshot 2023-12-03 at 6 53 11 AM" src="https://github.com/btcpayserver/btcpayserver/assets/3018359/736de5e3-d5f4-4456-a034-0f2b3080260c">

<img width="943" alt="Screenshot 2023-12-03 at 6 53 32 AM" src="https://github.com/btcpayserver/btcpayserver/assets/3018359/cc432815-3570-4645-a616-f13e4e819d34">

### Are you sure this is a bug report?

- [X] I confirm this is a bug report

**Comment by vindard** [comment]
Just checking, was this issue addressed? The commit message from #5534 suggests it wasn't and I still can't see recent transactions on a keypad if I'm logged out

**Comment by leesalminen** [comment]
The change made here was to hide the recent transactions button if logged
out. Previously, the recent transactions button showed when logged out, but
never loaded (just spinner forever) any data.

**Comment by vindard** [comment]
> The change made here was to hide the recent transactions button if logged out. Previously, the recent transactions button showed when logged out, but never loaded (just spinner forever) any data.

Ah ok gotcha. So for now we still need to be logged in for this to work properly 👍🏾

**Comment by dennisreimann** [comment]
Yes, we decided to require login for now, because we do not want to expose that data to anonymous users. 

We might adress this with something like login codes for a cashier role though.
